### PR TITLE
docs: add WASM runtime build instructions

### PR DESCRIPTION
**Pull request by tkennedy1-godaddy**
Commit Message: docs: add WASM runtime build instructions
Additional Description: This adds the specific language for enabling an optional WASM runtime for envoy to the bazel build instructions
Risk Level: low
Testing: n/a
Docs Changes: this is only a change to docs
Release Notes: n/a
Platform Specific Features: n/a
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
copying out CI advice

```console

  Please fix your editor to ensure:

      - no trailing whitespace
      - no preceding mixed tabs/spaces
      - all files end with a newline

```

File: bazel/README.md [hunk 1]
```diff
@@ -708,6 +708,8 @@ The extensions enabled by default can be disabled by adding the following parame
 `envoy.wasm.runtime.v8` extension, add `--//source/extensions/wasm_runtime/v8:enabled=false`.
 Note not all extensions can be disabled.
 
+To enable a specific wasm runtime, you'll need to pass `--define wasm=[wasm_runtime]`, e.g. `--define wasm=wasmtime` to enable the [wasmtime](https://wasmtime.dev/) runtime. The list of runtimes available can be found in the [bazel/BUILD](https://github.com/envoyproxy/envoy/blob/0a587f23b21ab34b28d9c53428af259b3f7b0970/bazel/BUILD#L407-L430) file.
```
Comment (PiotrSikora) [inline bazel/README.md:711]: I'd suggest replacing first the "wasm" with "WebAssembly (Wasm)" to help with discovery when grepping, etc.

Also, we've switched from "runtime" to "engine" in Proxy-Wasm (https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/256), since we're using custom Proxy-Wasm runtime envioronment, and not any provided by those projects (which often bundle both: engine and runtime).

Comment (PiotrSikora) [reply bazel/README.md:711]: Oh, and you should list available values here instead of linking to a specific commit, since it's going to get outdated.

File: external [hunk 2]
```diff
@@ -0,0 +1 @@
+bazel-out/../../../external
```
Comment (PiotrSikora) [inline external:1]: Unrelated change, please remove.

**Comment by tkennedy1-godaddy** [comment]
Thank you for the feedback -- I've made the changes.

I'm about to be out on vacation for a week so if there are more changes to get to, please let me know and I'll get to them when I'm back next week.

File: bazel/README.md [hunk 3]
```diff
@@ -708,6 +708,13 @@ The extensions enabled by default can be disabled by adding the following parame
 `envoy.wasm.runtime.v8` extension, add `--//source/extensions/wasm_runtime/v8:enabled=false`.
 Note not all extensions can be disabled.
 
+To enable a specific WebAssembly (Wasm) engine, you'll need to pass `--define wasm=[wasm_engine]`, e.g. `--define wasm=wasmtime` to enable the [wasmtime](https://wasmtime.dev/) engine. Supported engines are:
+
+* v8 (the default included engine)
+* wavm
+* wasmtime
+* wavr
```
Comment (PiotrSikora) [inline bazel/README.md:716]: 1. Please add backticks (e.g. wasmtime -> `wasmtime`).
2. `wavr` -> `wamr`.
3. It might make sense to sort the values alphabetically.

File: bazel/README.md [hunk 4]
```diff
@@ -991,3 +998,4 @@ Adding the following parameter to Bazel everytime or persist them in `.bazelrc`.
 ```
 --remote_http_cache=http://127.0.0.1:28080/
 ```
+
```
Comment (PiotrSikora) [inline bazel/README.md:1001]: Nit: unrelated change.

**Comment by tkennedy1-godaddy** [comment]
Thank you for your patience! I'm back and have updated with the requested changes

**Review by phlax — APPROVED** [review]
lgtm, thanks @tkennedy1-godaddy
